src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    public function execute(string &$query, array &$params = []): void
+    {
+        $finalParams = [];
+
+        foreach ($params as $paramKey => $paramValue) {
+            if (!is_array($paramValue)) {
+                $finalParams[$paramKey] = $paramValue;
+                continue;
+            }
+
+            if (is_numeric($paramKey)) {
+                throw new InvalidQueryException(sprintf('Unnamed parameter "%s" can\'t have array value.', $paramKey));
+            }
+
+            $valuesIndexedByPlaceholder = $this->getValuesIndexedByPlaceholder($paramKey, $paramValue);
```

Comment:
I would clean all of this up. It's a bit hard to understand what is going on without really looking at line by line. I suggest extracting to a few private methods so the names can explain what is being done.

Reply:
I just implemented a method to follow your suggestion:

```
transformQuery(string $query, string $currentPlaceholder, array $newPlaceholders): string
```

Do you see any other method that should be created?

Reply:
Seems better now.